File: fdlsandbox-android-java/app/src/main/java/com/omatt/fdlsandbox/firebase/AppInviteHelper.java
package com.omatt.fdlsandbox.firebase;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

import com.google.android.gms.appinvite.AppInviteInvitation;
import com.omatt.fdlsandbox.R;

/**
 * Created by omarmatthew on 6/25/17.
 * Class containing all necessary methods that uses Firebase App Invites
 */

public class AppInviteHelper {
    private final String TAG = "AppInviteHelper";
    /**
     * Returns Invite build template
     * @param context Context
     * @return Intent
     */
    public Intent appInviteTemplate(Context context) {
        return new AppInviteInvitation.IntentBuilder(context.getString(R.string.invitation_title))
                .setMessage(context.getString(R.string.invitation_message))
                .setDeepLink(Uri.parse(context.getString(R.string.invitation_deep_link)))
                .setCustomImage(Uri.parse(context.getString(R.string.invitation_custom_image)))
                .setCallToActionText(context.getString(R.string.invitation_cta))
                .build();
    }
}

File: fdlsandbox-android-kotlin/app/src/main/java/com/omatt/fdlsandbox/firebase/AppInviteHelper.kt
package com.omatt.fdlsandbox.firebase

import android.content.Context
import android.content.Intent
import android.net.Uri
import com.google.android.gms.appinvite.AppInviteInvitation
import com.omatt.fdlsandbox.R

/**
 * Created by omarmatthew on 9/16/2017.
 */
class AppInviteHelper {
    /**
     * Returns Invite build template
     * @param context Context
     * @param deepLink String
     * @return Intent
     */
    fun appInviteTemplate(context : Context, deepLink : String) : Intent {
//        return AppInviteInvitation.IntentBuilder(context.getString(R.string.invitation_title))
//                .setDeepLink(Uri.parse(deepLink))
//                .build()
        return AppInviteInvitation.IntentBuilder(context.getString(R.string.invitation_title))
                .setMessage(context.getString(R.string.invitation_message))
                .setDeepLink(Uri.parse(deepLink))
                .setCustomImage(Uri.parse(context.getString(R.string.invitation_custom_image)))
                .setCallToActionText(context.getString(R.string.invitation_cta))
                .build()
    }
}